tests/page/matchers.misc.spec.ts

```diff
@@ -23,3 +24,41 @@ it('should outlive frame navigation', async ({ page, server }) => {
   }, 1000);
   await expect(page.locator('.box').first()).toBeEmpty();
 });
+
+it('should print no-locator-resolved error when locator matcher did not resolve to any element', async ({ page, server }) => {
+  const myLocator = page.locator('.nonexisting');
+  const expectWithShortLivingTimeout = expect.configure({ timeout: 10 });
+  const locatorMatchers = [
```

Comment:
This test is not checking `.not.toHaveText()` that has changed extensively in the code. I'd recommend trimming this list to just a few, but add `.not` and any other interesting variants.

Reply:
Would you keep: one boolean matcher and one text matcher for each branch?

Reply:
Yeah, we should cover every branch, those are easy to mess up.